Submitting an issue should display a link to dashboard / redirect user to the dashboard

AlexGopal opened the pull request:
“Added redirect button on submission success screen to view created submission”

dependabot[bot] committed:
Added redirect button with submissionId integration

tallcoleman commented:
Thanks for doing this @AlexGopal , I'll give it a review as soon as I can!

tallcoleman commented:
@AlexGopal I'm having trouble telling which changes are formatting and which are functional. Could you please either:

- rebase to split into two commits, one for formatting changes, and one for functional
- note the files that have functional changes

Also, what formatting plugins and settings do you have in your editor? I think we would want to make sure that the formatting config in the repo is specific enough that we don't end up with duelling format changes down the line.

AlexGopal committed:
Applied Prettier formatting

AlexGopal commented:
I did the Rebase I think my prettier settings were different so i tried to set it to use project settings. 
Prettier was what i used for the formatting changes. I can remove the prettier changes and I think next time I run prettier it should be using the project settings. 
The first commit is the functional changes.

tallcoleman commented on bikespace_frontend/src/components/submission/summary/summary.module.scss:
```diff
@@ -3,4 +3,20 @@
\ No newline at end of file
+}
+
+.viewButton {
```
A couple things to consider for this button:

- For the colors etc. we have a variables file you can import with `@import '@/styles/variables.module.scss';` at the top and then you can use the brand colors via their variable references.
- The "main" brand green [doesn't provide enough contrast for white text](https://webaim.org/resources/contrastchecker/?fcolor=FFFFFF&bcolor=4CAF50), so we usually use a darkened version instead (e.g. `$color-primary-d50p` or `$main-color-bg`)
- There are some other existing button elements you could re-use, e.g.
  - SidebarButton from `@/components/shared-ui/sidebar-button`
  - You could pull out the buttons in `@/components/submission/submission-form-controller` into a shared UI folder within `@/components/submission/` and use that here

AlexGopal replied:
Okay sure, I will attempt these changes and let you know,  sorry for the late response I was going to try it then respond but I got a bit busy so I'll try it and let you know.

tallcoleman replied:
Sounds good, thanks!

AlexGopal replied:
So I'm curious what do you prefer should I use SidebarButton or should I maybe try pulling out the buttons from @/components/submission/submission-form-controller and using it here. 

The second one is harder but I'm willing to try. 

I attached two photos the first is using Sidebarbutton, the second is using /components/submission/submission-form-controller.

Which one do you prefer?

![sidebarbutton](https://github.com/user-attachments/assets/8d5a2800-baa7-44fb-a563-25bb1d34ed2c)
![submission_form_controller_button](https://github.com/user-attachments/assets/91077a18-4760-4599-91dc-1387b455ec2d)

tallcoleman replied:
Hi @AlexGopal 

I asked for some advice on this since some of the other designers have been talking about wanting to harmonize some of the UI between the pages. I think either of the following would be fine:

1. Keep it consistent with the existing `/submission` button styling (like your second photo) and don't worry too much about code re-use, since we may end up re-doing all the button designs in the submission form later. In this case, copy-pasting code and styling from other parts of the submission form would be totally OK, as long as the text contrast is good (a decent hover state would also be nice, but not necessary). 
2. Use SidebarButton for the "View Your Submission" button and also update the form control buttons (e.g. "Next", "Back", "Close" to use SidebarButton as well for consistency. The other input controls can be left as-is.

tallcoleman reviewed (changes requested):
This is really great, thanks @AlexGopal !

Just one requested change to think about code re-use and accessibility for the button - if you have any questions about how to resolve, just let me know.

AlexGopal committed:
fixed button to be consistent with the current submission button styling

tallcoleman commented on bikespace_frontend/src/components/submission/summary/Summary.tsx:
```diff
@@ -1,6 +1,9 @@
+import Link from 'next/link';
 
 import styles from './summary.module.scss';
+import {SidebarButton} from '@/components/shared-ui/sidebar-button';
```
Can remove SidebarButton import since it's not being used

tallcoleman commented on bikespace_frontend/src/components/submission/summary/Summary.tsx:
```diff
@@ -39,6 +43,20 @@ export const Summary = () => {
+
+          {submissionId ? (
+            <Link href={`/dashboard?tab=feed&submission_id=${submissionId}`}>
+              <button className={submissionStyles.primaryBtn}>
```
Please add to the button attributes `data-umami-event="view-issue-from-submission-confirmation"`

tallcoleman commented on bikespace_frontend/src/components/submission/summary/Summary.tsx:
```diff
@@ -39,6 +43,20 @@ export const Summary = () => {
+            </Link>
+          ) : (
+            <Link href="/dashboard">
+              <button className={submissionStyles.primaryBtn}>
```
Please add to the button attributes `data-umami-event="view-dashboard-from-submission-confirmation"`

tallcoleman committed:
Merge branch 'main' into pr/290

tallcoleman committed:
Add coverage to response object from submission

tallcoleman reviewed (changes requested):
Thanks @AlexGopal this looks good, and looking forward to getting it merged soon!

Have left a couple comments for final tweaks (e.g. adding analytics properties for the buttons). I also caught the branch up to main and added test coverage on the API side to ensure the submission id is part of the response object for a successful issue report.

It would also be helpful to add test coverage in `bikespace_frontend/src/components/submission/summary/summary.test.tsx` (I think the key thing would be ensuring that the URL is properly formed). 

You can let us know whether you want to give this a go or want to get some help; if you're not familiar with jest and react-testing-library, it can sometimes be a bit frustrating.

AlexGopal committed:
removed unused import and added to button attributes

AlexGopal committed:
Add frontend test for submission dashboard URL to ensure url is properly formatted.

tallcoleman committed:
Merge branch 'main' into pr/290

tallcoleman committed:
Set different umami event if fallback post-submission link is used

tallcoleman committed:
Change submission e2e test to include View Your Submission button

- added timestamp to test comment to ensure test can be run back to back when testing locally (otherwise duplicate comments throw an error until the database is fully re-started)
- fixed bug where sidebar did not open on first load on mobile when a feed param is specified

tallcoleman reviewed (approved):
Thanks again @AlexGopal ! This is ready to go now - will approve and merge.
